helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1256,14 +1256,18 @@ protected void checkLiveInstancesObservation(List<LiveInstance> liveInstances,
 
         for (String instance : curInstances.keySet()) {
           if (lastInstances == null || !lastInstances.containsKey(instance)) {
-            try {
-              manager.addCustomizedStateRootChangeListener(this, instance);
-              logger.info(manager.getInstanceName() + " added customized root change listener for"
-                  + " " + instance
-                  + ", listener: " + this);
-            } catch (Exception e) {
-              logger.error("Fail to add customized root change listener for instance: " + instance,
-                  e);
+            PropertyKey propertyKey =
+                keyBuilder.customizedStatesRoot(instance);
+            if (propertyKey.getPath() != null) {
```

Comment:
You're creating a PropertyKey object with the given instance name and checking whether it's null? How would it be null if you've just created the path?

Did you mean to check whether the path exists in ZooKeeper? If so, shouldn't you do an exist() check with a ZkClient-based API?

Unless I'm mistaken and we actually somehow pre-populate the KeyBuilder with a whitelist of instances for the CustomizedView feature?

Reply:
oh..I was thinking to use exist checking, but put the wrong check, let me update it.
This is just for backward incompatibility check. We're totally fine without this check, but the participant would need to be on the latest version to have customized state path created. If they don't, there can be some error logs when adding listener to customized state path. It's still fine, but might cause confusion.

Reply:
Yes, for backward compatibility, you need to check the root path existence in ZK. This check is not very meaningful. You're basically doing `if ("~~/instanceName" != null)`.